x-pack/plugins/security_solution/server/search_strategy/timeline/factory/details/index.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getOr, merge } from 'lodash/fp';
+
+import { IEsSearchResponse } from '../../../../../../../../src/plugins/data/common';
+
+import {
+  FactoryQueryTypes,
+  TimelineDetailsStrategyResponse,
+  TimelineDetailsQueries,
+  TimelineDetailsRequestOptions,
+} from '../../../../../common/search_strategy/timeline';
+
+import { inspectStringifyObject } from '../../../../utils/build_query';
+import { SecuritySolutionTimelineFactory } from '../types';
+import { buildTimelineDetailsQuery } from './dsl/query.timeline_details.dsl';
+import { getDataFromHits } from './helpers';
+
+export const timelineDetails: SecuritySolutionTimelineFactory<TimelineDetailsQueries.timelineDetails> = {
+  buildDsl: (options: TimelineDetailsRequestOptions) => {
+    const { indexName, eventId, docValueFields = [] } = options;
+    return buildTimelineDetailsQuery(indexName, eventId, docValueFields);
+  },
+  parse: async (
+    options: TimelineDetailsRequestOptions,
+    response: IEsSearchResponse<unknown>
+  ): Promise<TimelineDetailsStrategyResponse> => {
+    const { indexName, eventId, docValueFields = [] } = options;
+    const sourceData = getOr({}, 'hits.hits.0._source', response.rawResponse);
+    const hitsData = getOr({}, 'hits.hits.0', response.rawResponse);
+    delete hitsData._source;
+    const inspect = {
+      dsl: [inspectStringifyObject(buildTimelineDetailsQuery(indexName, eventId, docValueFields))],
+      response: [inspectStringifyObject(response.rawResponse)],
```

Comment:
we could safely remove the response here
https://github.com/elastic/kibana/pull/76090#discussion_r480130677